### PR TITLE
[MERGE] mail, sms: apply various fixes

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
PURPOSE

Fix issues spotted when testing sms

SPECIFICATIONS

See sub commits for more details. In short

  * crm: allow phone widget to display SMS composer icon;
  * sms: fix MRO issue on res.partner;
  * test_mail: fix some search / comparisons not completely deterministic;
  * website_sale: fix mail test relying on demo data;

LINKS

Task 2060220

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

File: addons/sms/models/res_partner.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class ResPartner(models.Model):
     _name = 'res.partner'
-    _inherit = ['res.partner', 'mail.thread.phone']
+    _inherit = ['mail.thread.phone', 'res.partner']
```
Comment (tivisse) [inline addons/sms/models/res_partner.py]: @rco-odoo This is the issue I was talking about yesterday.

**Pull request closed without merging**